gno/r/launchpad_grc20/sale_grc20.gno

```diff
@@ -0,0 +1,269 @@
+package launchpad_grc20
+
+import (
+	"encoding/hex"
+	"std"
+	"time"
+
+	"gno.land/p/demo/avl"
+	"gno.land/p/demo/json"
+	"gno.land/p/demo/merkle"
+	"gno.land/p/demo/seqid"
+	"gno.land/p/demo/ufmt"
+	"gno.land/p/teritori/jsonutil"
+)
+
+type Sale struct {
+	token          *Token
+	startTimestamp int64
+	endTimestamp   int64
+	pricePerToken  uint64
+	alreadySold    uint64
+	limitPerAddr   uint64
+	minGoal        uint64
+	maxGoal        uint64
+	owner          std.Address
+	buyers         *avl.Tree // address -> amount
+	finalized      bool
+	merkleRoot     string
+}
+
+var (
+	sales      *avl.Tree // sale ID -> sale
+	nextSaleID seqid.ID
+)
+
+func init() {
```

Comment:
Didn't know you can do multiple inits inside of the same package. Cool. I think it might be worthwhile taking a minute to check out the initialization order, maybe this could create some bugs down the line for users.

Reply:
Yes, the order is the alphabetical order so airdrop, then sale and factory (it works like this for go at least), anyway the init are not relying on each other so does not matter the order of execution i think